ingest/scripts/group_segments.py

```diff
@@ -94,27 +108,38 @@ def main(
     # Group segments according to isolate, collection date and isolate specific values
     # These are the fields that are expected to be identical across all segments for a given isolate
 
-    first_row = next(iter(segment_metadata.values()))
-    if not first_row:
+    # Dynamically determine the fields that are present in the metadata
+    some_row = next(iter(segment_metadata.values()))
```

Comment:
Is this _not_ the first row? If it is still that then I find the original version clearer fwiw (though I found your changes above very helpful)

Reply:
Since we're using Python >=3.7 there _is_ a first element, so you're right!

https://mail.python.org/pipermail/python-dev/2017-December/151283.html